### PR TITLE
[Feature]add cn into default warehouse

### PR DESCRIPTION
**Pull request by abc982627271**
## What type of PR is this：
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 3 / 37 (08.11%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/system/SystemInfoService.java|0|17|00.00%|[174, 175, 176, 177, 339, 340, 341, 342, 1017, 1018, 1019, 1020, 1055, 1069, 1070, 1071, 1072]|
|:large_blue_circle:|com/starrocks/warehouse/LocalWarehouse.java|0|1|00.00%|[56]|
|:large_blue_circle:|com/starrocks/warehouse/Cluster.java|1|11|09.09%|[67, 68, 69, 70, 73, 76, 77, 78, 79, 81]|
|:large_blue_circle:|com/starrocks/system/ComputeNode.java|1|7|14.29%|[148, 149, 152, 156, 157, 160]|
|:large_blue_circle:|com/starrocks/system/HeartbeatMgr.java|1|1|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/system/ComputeNode.java [hunk 1]
```diff
@@ -94,6 +94,13 @@ public class ComputeNode implements IComputable, Writable {
     @SerializedName("lastWriteFail")
     private volatile boolean lastWriteFail = false;
 
+    // for shared_data runmode
+    @SerializedName("warehouseName")
+    private String warehouseName;
```
Comment (sduzh) [inline fe/fe-core/src/main/java/com/starrocks/system/ComputeNode.java:99]: Is it possible to change the name of the warehouse in the future?

File: fe/fe-core/src/main/java/com/starrocks/system/ComputeNode.java [hunk 2]
```diff
@@ -94,6 +94,13 @@ public class ComputeNode implements IComputable, Writable {
...
+
+    @SerializedName("workerGroupId")
+    private long workerGroupId = -1;
```
Comment (sduzh) [inline fe/fe-core/src/main/java/com/starrocks/system/ComputeNode.java:102]: Can we save the workerGroupId in Warehouse instead of the ComputeNode?

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 5 / 28 (17.86%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/system/SystemInfoService.java|0|17|00.00%|[174, 175, 176, 177, 339, 340, 341, 342, 1017, 1018, 1019, 1020, 1055, 1069, 1070, 1071, 1072]|
|:large_blue_circle:|com/starrocks/warehouse/Cluster.java|1|5|20.00%|[66, 67, 70, 71]|
|:large_blue_circle:|com/starrocks/system/HeartbeatMgr.java|4|6|66.67%|[247, 248]|

File: fe/fe-core/src/main/java/com/starrocks/warehouse/Cluster.java [hunk 3]
```diff
@@ -20,16 +20,21 @@
...
+import java.util.ArrayList;
+import java.util.List;
 
 public class Cluster implements Writable {
     @SerializedName(value = "id")
     private long id;
     @SerializedName(value = "wgid")
     private long workerGroupId;
+    @SerializedName(value = "computeNodeList")
+    private List<ComputeNode> computeNodeList = new ArrayList<>();
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/warehouse/Cluster.java:37]: why not use `map<id, cn>` or `set<id>`?

File: fe/fe-core/src/main/java/com/starrocks/warehouse/Cluster.java [hunk 4]
```diff
@@ -24,12 +24,16 @@
...
+import java.util.HashSet;
+import java.util.Set;
 
 public class Cluster implements Writable {
     @SerializedName(value = "id")
     private long id;
     @SerializedName(value = "wgid")
     private long workerGroupId;
+    @SerializedName(value = "computeNodeList")
+    private Set<Long> computeNodeIdList = new HashSet<>();
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/warehouse/Cluster.java:36]:
```suggestion
    @SerializedName(value = "computeNodeIds")
    private Set<Long> computeNodeIds = new HashSet<>();
```

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 5 / 28 (17.86%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/system/SystemInfoService.java|0|17|00.00%|[174, 175, 176, 177, 339, 340, 341, 342, 1017, 1018, 1019, 1020, 1055, 1068, 1069, 1070, 1071]|
|:large_blue_circle:|com/starrocks/warehouse/Cluster.java|1|5|20.00%|[65, 66, 69, 70]|
|:large_blue_circle:|com/starrocks/system/HeartbeatMgr.java|4|6|66.67%|[247, 248]|

**Review by wyb — APPROVED** [review]

**Review by gengjun-git — APPROVED** [review]

**Pull request merged**